cmd/client-s3.go

```diff
@@ -2120,57 +2131,61 @@ func (c *S3Client) ShareUpload(isRecursive bool, expires time.Duration, contentT
 }
 
 // SetObjectLockConfig - Set object lock configurataion of bucket.
-func (c *S3Client) SetObjectLockConfig(mode *minio.RetentionMode, validity *uint, unit *minio.ValidityUnit) *probe.Error {
+func (c *S3Client) SetObjectLockConfig(mode minio.RetentionMode, validity uint64, unit minio.ValidityUnit) *probe.Error {
 	bucket, _ := c.url2BucketAndObject()
 
-	err := c.api.SetBucketObjectLockConfig(bucket, mode, validity, unit)
-	if err != nil {
-		return probe.NewError(err)
+	// FIXME: This is too ugly, fix minio-go
+	vuint := (uint)(validity)
```

Comment:
```
var vuint *uint
	if validity != 0 {
		vuint = nil
	}
	var modePtr *minio.RetentionMode
	if mode.IsValid() {
		modePtr = &mode
	}
	var unitPtr *minio.ValidityUnit
	if unit == minio.Days || unit == minio.Years {
		unitPtr = &unit
	}
	e := c.api.SetBucketObjectLockConfig(bucket, modePtr, vuint, unitPtr)

```
Should be something like this. Minio-go rejects if the value is not valid

Reply:
fixed